packages/api-request-builder/src/default-params.js

```diff
@@ -62,23 +63,104 @@ export function getDefaultSearchParams (): ServiceBuilderDefaultParams {
 export function setDefaultParams () {
   this.params.expand = getDefaultQueryParams().expand
 
-  if (this.features.indexOf(features.queryOne) >= 0)
+  if (this.features.includes(features.queryOne))
     this.params.id = getDefaultQueryParams().id
 
-  if (this.features.indexOf(features.query) >= 0) {
+  if (this.features.includes(features.query)) {
     this.params.pagination = getDefaultQueryParams().pagination
     this.params.query = getDefaultQueryParams().query
   }
 
-  if (this.features.indexOf(features.search) >= 0) {
+  if (this.features.includes(features.search)) {
     this.params.staged = getDefaultSearchParams().staged
     this.params.pagination = getDefaultSearchParams().pagination
     this.params.search = getDefaultSearchParams().search
   }
 
-  if (this.features.indexOf(features.projection) >= 0)
+  if (this.features.includes(features.projection))
     this.params.staged = true
 
-  if (this.features.indexOf(features.suggest) >= 0)
+  if (this.features.includes(features.suggest))
     this.params.searchKeywords = []
 }
+
+const hasKey = (obj: {}, key: string): boolean =>
+  Object.prototype.hasOwnProperty.call(obj, key)
+
+/**
+ * Set the supplied parameters given the current service object.
+ *
+ * @return {void}
+ */
+export function setParams (params: ServiceBuilderParams) {
+  // verify params
+  const knownKeys = [
+    'expand', 'id', 'key', 'customerId', 'sort', 'page', 'perPage', 'staged',
+    'priceCurrency', 'priceCountry', 'priceCustomerGroup', 'priceChannel',
+    'text', 'fuzzy', 'fuzzyLevel', 'markMatchingVariants', 'facet', 'filter',
+    'filterByQuery', 'filterByFacets', 'searchKeywords', 'where', 'version',
+  ]
+  Object.keys(params).forEach((key: string) => {
+    if (!knownKeys.includes(key)) throw new Error(`Unknown key "${key}"`)
+  })
+
+  // query-expand
+  if (params.expand)
+    params.expand.forEach((expansion: string) => { this.expand(expansion) })
+
+  // query-id
+  if (hasKey(params, 'id')) this.byId(params.id)
+  if (hasKey(params, 'key')) this.byKey(params.key)
+  if (hasKey(params, 'customerId')) this.byCustomerId(params.customerId)
+
+  // query-page
+  if (params.sort) params.sort.forEach(
```

Comment:
The internal representation nests these objects as `pagination: { page, perPage, sort }`. The public API has no nesting. Not sure which is better.